File: src/shared/BasicCommerce.Domain/Entities/Terminal.cs
using BasicCommerce.Domain.Enums;

namespace BasicCommerce.Domain.Entities;

public class Terminal : TenantEntity
{
    public Guid StoreId { get; private set; }
    public string Name { get; private set; } = default!;
    public string Code { get; private set; } = default!;
    public TerminalType Type { get; private set; }
    public TerminalStatus TerminalStatus { get; private set; } = TerminalStatus.Offline;
    public Guid? CurrentCashierId { get; private set; }
    public decimal OpeningFloat { get; private set; }
    public DateTime? LastActivityAt { get; private set; }


    private Terminal() { }

    public static Terminal Create(Guid tenantId, Guid storeId, string name,
        string code, TerminalType type = TerminalType.Standard)
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(name);
        ArgumentException.ThrowIfNullOrWhiteSpace(code);

        return new Terminal
        {
            TenantId = tenantId,
            StoreId = storeId,
            Name = name,
            Code = code.ToUpperInvariant(),
            Type = type
        };
    }

    public void OpenShift(Guid cashierId, decimal openingFloat)
    {
        CurrentCashierId = cashierId;
        OpeningFloat = openingFloat;
        TerminalStatus = TerminalStatus.Online;
        LastActivityAt = DateTime.UtcNow;
    }

    public void CloseShift()
    {
        CurrentCashierId = null;
        OpeningFloat = 0;
        TerminalStatus = TerminalStatus.Offline;
    }

    public void RecordActivity() => LastActivityAt = DateTime.UtcNow;

    public void Deactivate()
    {
        Status = EntityStatus.Inactive;
        UpdatedAt = DateTime.UtcNow;
    }
}

File: src/clients/BasicCommerce.WebAdmin/src/types/terminal.ts
export const TERMINAL_TYPES = ['Standard', 'SelfCheckout', 'MobilePOS', 'KioskOrder'] as const;
export type TerminalType = (typeof TERMINAL_TYPES)[number];

export interface Terminal {
  id: string;
  storeId: string;
  storeName: string;
  name: string;
  code: string;
  type: string;
  status: string;
  operationalStatus: string;
  currentCashierName: string | null;
}

export interface TerminalFormData {
  storeId: string;
  name: string;
  code: string;
  type: TerminalType;
}
